src/pages/Dashboard/components/DashboardHeader/DashboardHeader.tsx

```diff
@@ -0,0 +1,209 @@
+import { memo, useEffect, useLayoutEffect, useMemo, useRef } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useLocation } from 'react-router'
+import { Amount } from 'components/Amount/Amount'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { useProfileAvatar } from 'hooks/useProfileAvatar/useProfileAvatar'
+import {
+  selectClaimableRewards,
+  selectPortfolioTotalUserCurrencyBalanceExcludeEarnDupes,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { DashboardTab } from '../DashboardTab'
+import { DashboardHeaderTop } from './DashboardHeaderTop'
+import { EarnBalance } from './EarnBalance'
+
+const paddingTop = { base: 'env(safe-area-inset-top)', md: '4.5rem' }
+const marginTop = { base: 0, md: '-4.5rem' }
+
+export type TabItem = {
+  label: string
+  path: string
+  color: string
+  exact?: boolean
+  rightElement?: JSX.Element
+  hide?: boolean
+}
+
+const flexDirTabs: ResponsiveValue<Property.FlexDirection> = { base: 'column', md: 'row' }
+const navItemPadding = { base: 6, '2xl': 8 }
+const navCss = {
+  '&::-webkit-scrollbar': {
+    display: 'none',
+  },
+}
+
+const headerPosition: ResponsiveValue<Property.Position> = { base: 'fixed', md: 'relative' }
+
+export const DashboardHeader = memo(({ tabComponent }: { tabComponent?: React.ReactNode }) => {
+  const isNftsEnabled = useFeatureFlag('Jaypegz')
+  const location = useLocation()
+  const avatarImage = useProfileAvatar()
+  const translate = useTranslate()
+  const activeRef = useRef<HTMLButtonElement | null>(null)
+  const containerRef = useRef<HTMLDivElement | null>(null)
+  const claimableRewardsUserCurrencyBalanceFilter = useMemo(() => ({}), [])
+  const claimableRewardsUserCurrencyBalance = useAppSelector(state =>
+    selectClaimableRewards(state, claimableRewardsUserCurrencyBalanceFilter),
+  )
+  const portfolioTotalUserCurrencyBalance = useAppSelector(
+    selectPortfolioTotalUserCurrencyBalanceExcludeEarnDupes,
+  )
+  const borderColor = useColorModeValue('gray.100', 'whiteAlpha.200')
+
+  useEffect(() => {
+    if (activeRef.current) {
+      activeRef.current.scrollIntoView({ behavior: 'smooth', block: 'center', inline: 'center' })
+    }
+  }, [location])
+
+  const NavItems: TabItem[] = useMemo(() => {
+    return [
+      {
+        label: 'common.overview',
+        path: '/wallet',
+        color: 'blue',
+        exact: true,
+      },
+      {
+        label: 'navBar.wallet',
+        path: '/wallet/accounts',
+        color: 'blue',
+        rightElement: <Amount.Fiat value={portfolioTotalUserCurrencyBalance} />,
+      },
+      {
+        label: 'navBar.defi',
+        path: '/wallet/earn',
+        color: 'purple',
+        rightElement: <EarnBalance />,
+      },
+      {
+        label: 'navBar.rewards',
+        path: '/wallet/rewards',
+        color: 'green',
+        rightElement: <Amount.Fiat value={claimableRewardsUserCurrencyBalance} />,
```

Comment:
These `rightElement` depend on dynamic redux data which triggers this whole object to re-instantiate whenever the deps change. Instead, they should be self-contained components so we can make NavItems static without useMemo, thereby preventing a cascade of compute whenever the portfolio / claimable rewards changes